app/models/concerns/facets/base.rb

```diff
@@ -42,7 +42,7 @@ def inherited_attributes(hostgroup, facet_attributes)
         _, facet_config = Facets.find_facet_by_class(self, :host)
 
         if facet_config.has_hostgroup_configuration? && hostgroup
-          facet_attributes = hostgroup.inherited_facet_attributes(facet_config)
+          facet_attributes = hostgroup.inherited_facet_attributes(facet_config).merge(facet_attributes || {}) { |_, left, right| right || left }
```

Comment:
Before taking the `left` part, you can test if the key is present in the `facet_attributes`. This will enable you to set a nil value for a specific host. It would be consistent with the rest of the facets infrastructure.

e.g.
``` ruby
facet_attributes = { a: nil }
hg_attributes = { a: 'inherited_value' }
# will result in:
to_apply = { a: nil }

# But on the other hand
facet_attributes = { } # :a is not specified at all
hg_attributes = { a: 'inherited_value' }
# will result in:
to_apply = { a: 'inherited_value' }
```